### DISCUSSION TITLE
How to use range filtering with byte (varbinary(16)) in MySQL

### DISCUSSION THREAD

**Discussion by dohankk** [prisma/prisma#21444] [Q&A] [answered]
### Question

```prisma
model Users {
    id        Bytes    @id @db.VarBinary(16)
    name      String   @db.VarChar(100)
    createdAt DateTime @default(now()) @map("created_at")

    @@map("users")
}

```
when id is varbinary(16) type, i want to use range filter.

```ts
const users = await prisma.users.findMany({
  where: {
   id: {
      lte: Buffer.from(UUID_STRING),
    }
  }
});
```

but `BytesFilter` does not have `gte` `lte`

so i use like below

```js
const users = await prisma.$queryRaw`select * from users where id > UUID_TO_BIN(${uuid})`;
```

how can i use range filter?

**Comment by ludralph** [comment, marked as answer]
Hi @dohankk 👋 

Thank you for raising this question.

Prisma currently does not support range filters like `lte` or `gte` for the `Bytes` type. The `Bytes` type can be filtered with `in` and `notIn` operations as per the [Prisma blog post](https://www.prisma.io/blog/wnip-q4-dsk0golh8v#bytes-can-now-be-filtered-with-in-and-notin). However, for more complex filtering, you might need to use raw SQL queries with Prisma.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
